lib/rucio/tests/test_undertaker.py

```diff
@@ -15,22 +15,33 @@
 
 from datetime import datetime, timedelta
 from logging import getLogger
+from collections import namedtuple
 
 import pytest
 
 from rucio.common.policy import get_policy
 from rucio.common.types import InternalAccount, InternalScope
 from rucio.core.account_limit import set_local_account_limit
-from rucio.core.did import add_dids, attach_dids, list_expired_dids, get_did, set_metadata
+from rucio.core.did import (
+    add_dids, attach_dids, list_expired_dids, get_did, set_metadata,
+    list_content, list_all_parent_dids
+)
 from rucio.core.replica import add_replicas, get_replica
 from rucio.core.rse import get_rse_id, add_rse
-from rucio.core.rule import add_rules, list_rules
+from rucio.core.rule import add_rules, add_rule, list_rules
 from rucio.daemons.judge.cleaner import rule_cleaner
 from rucio.daemons.undertaker.undertaker import undertaker
 from rucio.db.sqla.util import json_implemented
+from rucio.db.sqla.constants import OBSOLETE
 from rucio.tests.common import rse_name_generator, did_name_generator
+from rucio.common.exception import DataIdentifierNotFound
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from rucio.tests.temp_factories import TemporaryRSEFactory, TemporaryDidFactory
 
 LOG = getLogger(__name__)
+Rse = namedtuple('Rse', ['name', 'id'])
```

Comment:
`Rse`  capitalisation looks... very wrong

Reply:
true. Changed it to `RSE_namedtuple`